code/modules/mining/mine_turfs.dm

```diff
@@ -589,19 +590,83 @@ var/global/list/rockTurfEdgeCache
 			F.state = L.state
 		playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
 
+/obj/effect/overlay/sandpile
+	name = "sand pile"
+	icon = 'icons/turf/mining.dmi'
+	icon_state = "buried"
+	mouse_opacity = 0 //So we don't have to copy-paste digging code for the pile too
+	var/atom/movable/buried //Did we bury something in this pile?
+	var/turf/simulated/floor/plating/asteroid/owner
+
+/obj/effect/overlay/sandpile/Destroy()
+	if (istype(buried))
+		buried.forceMove(owner) //So that objects that are inside us aren't destroyed :p
+	if (istype(owner))
+		owner.pile = null
+	return ..()
+
+/obj/effect/overlay/sandpile/container_resist()
+	var/mob/living/user = usr
+	user.changeNext_move(CLICK_CD_BREAKOUT)
+	user.last_special = world.time + CLICK_CD_BREAKOUT
+	user << "<span class='notice'>You start violently digging your way out of the sand pile!</span>"
+	if(do_after(user, 600, target = src))
+		user << "<span class='notice'>You manage to dig your way out!</span>"
+		if (istype(owner))
+			user.forceMove(owner)
+			buried = null
+			owner.gets_dug()
+		else
+			qdel(src) //Failsafe
+
 /turf/simulated/floor/plating/asteroid/proc/gets_dug()
 	if(dug)
 		return
-	new/obj/item/weapon/ore/glass(src)
-	new/obj/item/weapon/ore/glass(src)
-	new/obj/item/weapon/ore/glass(src)
-	new/obj/item/weapon/ore/glass(src)
-	new/obj/item/weapon/ore/glass(src)
+	if(istype(pile))
+		qdel(pile)
+		pile = null
+
+	new/obj/item/weapon/ore/sand(src)
+	new/obj/item/weapon/ore/sand(src)
+	new/obj/item/weapon/ore/sand(src)
+	new/obj/item/weapon/ore/sand(src)
+	new/obj/item/weapon/ore/sand(src)
 	dug = 1
 	icon_plating = "asteroid_dug"
 	icon_state = "asteroid_dug"
 	return
 
+/turf/simulated/floor/plating/asteroid/MouseDrop_T(atom/movable/target, mob/user)
```

Comment:
Iunno wouldn't it be better to Buckle the mob to the turf and then deal with it from there